interface/overlaps.py

```diff
@@ -0,0 +1,48 @@
+import streamlit as st
+from annotated_text import annotated_text
+
+
+def st_annotate_highlights(text, highlights):
+    highlighted_text = text
+    for highlight in highlights:
+        # TODO should this be normalized to handle i.e. lower-case matches?
+        highlighted_text = highlighted_text.replace(
+            highlight,
+            "--hl--hl>>" + highlight + "<<hl--hl--"
+        )
+    
+    text_with_highlights = []
+    for fragment in highlighted_text.split("--hl--"):
+        if fragment.startswith("hl>>") and fragment.endswith("<<hl"):
+            sanitized_fragment = fragment.replace("hl>>", "").replace("<<hl", "")
+            text_with_highlights.append((sanitized_fragment, "highlight"))
```

Comment:
The `"highlight"` argument could potentially add more info about the type of highlight (ie NER):
![Screenshot 2022-04-14 at 14 47 51](https://user-images.githubusercontent.com/1349225/163456764-0be40fe7-4c3e-43cf-86f7-908600c8f3c6.png)

We can also add coloring if it's useful.

Reply:
yup